pkgs/build-support/cc-wrapper/add-flags.sh

```diff
@@ -1,37 +1,112 @@
+# N.B. It may be a surprise that the derivation-specific variables are exported,
+# since this is just sourced by the wrapped binaries---the end consumers. This
+# is because one wrapper binary may invoke another (e.g. cc invoking ld). In
+# that case, it is cheaper/better to not repeat this step and let the forked
+# wrapped binary just inherit the work of the forker's wrapper script.
+
+# Accumulate prefixes for taking in the right input parameters. See setup-hook
+# for details.
+declare -a role_prefixes=()
+if [[ -n "${NIX_CC_WRAPPER_@infixSalt@_TARGET_BUILD:-}" ]]; then
+    role_prefixes+=(_BUILD)
+fi
+if [[ -n "${NIX_CC_WRAPPER_@infixSalt@_TARGET_HOST:-}" ]]; then
+    role_prefixes+=('')
+fi
+if [[ -n "${NIX_CC_WRAPPER_@infixSalt@_TARGET_TARGET:-}" ]]; then
+    role_prefixes+=(_TARGET)
+fi
+
+# For each role we serve, we accumulate the input parameters into our own
+# cc-wrapper-derivation-specific environment variables.
+for pre in "${role_prefixes[@]}"; do
+    # We need to mangle names for hygiene, but also take parameters/overrides
+    # from the environment.
+    slurpUnsalted () {
+        case "$1" in
+            CC_WRAPPER_*)
+                local firstPre=NIX_CC_WRAPPER_
+                local varname="${1#CC_WRAPPER_}"
+                ;;
+            LD_WRAPPER_*)
+                local firstPre=NIX_LD_WRAPPER_
+                local varname="${1#LD_WRAPPER_}"
+                ;;
+            *)
+                local firstPre=NIX_
+                local varname="$1"
+                ;;
+        esac
+        local inputVar="${firstPre}${pre}${varname}"
+        local outputVar="${firstPre}@infixSalt@_${varname}"
+        local delimiter=''
+        if [[ -n "${!outputVar:-}" && -n "${!inputVar:-}" ]]; then
+            delimiter=' '
+        fi
+        # Easiest to just do this to deal with either the input or (old) output.
+        set +u
+        export ${outputVar}+="${delimiter}${!inputVar}"
```

Comment:
This appears to be setting the variables unconditionally, which broke #6688 (standalone `clang++`). I'm not sure what the exact interactions here are but `nix run clang` can't find the C++ standard library headers and I've tracked it down to this variable assignment.

Reply:
Probably the right fix is to use `${NIX_CXXSTDLIB_COMPILE:-@default_cxx_stdlib_compile@}` instead of `${NIX_CXXSTDLIB_COMPILE-@default_cxx_stdlib_compile@}`, in `cc-wrapper.sh` to treat the empty variable the same way as if it was uninitialized, IIRC.